drivers/include/net/netdev.h

```diff
@@ -390,6 +388,113 @@ typedef struct netdev_driver {
                const void *value, size_t value_len);
 } netdev_driver_t;
 
+/**
+ * @brief   Add a netdev layer to the netdev layer stack
+ *
+ * @param[in] top     Last netdev device added to the stack.
+ * @param[in] new     New layer to push to the top of the stack.
+ *
+ *
+ * @return  The new top netdev layer of the netdev stack.
+ */
+netdev_t *netdev_add_layer(netdev_t *top, netdev_t *dev);
+
+/**
+ * @brief   Passthrough init function.
+ *
+ * See also @ref netdev_driver for the extended description of this functions
+ * behaviour
+ *
+ * @param[in]   dev     network device descriptor
+ * @return `< 0` on error, 0 on success
+ */
+int netdev_init_pass(netdev_t *dev);
+
+/**
+ * @brief   Passthrough isr function.
+ *
+ * See also @ref netdev_driver for the extended description of this functions
+ * behaviour
+ *
+ * @param[in]   dev     network device descriptor
+ */
+void netdev_isr_pass(netdev_t *dev);
+
+/**
+ * @brief   Passthrough send function.
+ *
+ * See also @ref netdev_driver for the extended description of this functions
+ * behaviour
+ *
+ * @param[in] dev       network device descriptor
+ * @param[in] vector    io vector array to send
+ * @param[in] count     nr of entries in vector
+ *
+ * @return number of bytes sent, or `< 0` on error
+ */
+int netdev_send_pass(netdev_t *dev, const struct iovec *vector, unsigned count);
+
+/**
+ * @brief   Passthrough recv function.
+ *
+ * See also @ref netdev_driver for the extended description of this functions
+ * behaviour
+ *
+ * @param[in]   dev     network device descriptor
+ * @param[out]  buf     buffer to write into or NULL
+ * @param[in]   len     maximum number of bytes to read
+ * @param[out] info     status information for the received packet. Might
+ *                      be of different type for different netdev devices.
+ *                      May be NULL if not needed or applicable.
+ *
+ * @return `< 0` on error
+ * @return number of bytes read if buf != NULL
+ * @return packet size if buf == NULL
+ */
+int netdev_recv_pass(netdev_t *dev, void *buf, size_t len, void *info);
+
+/**
+ * @brief   Passthrough get function.
+ *
+ * See also @ref netdev_driver for the extended description of this functions
+ * behaviour
+ *
+ * @param[in]   dev     network device descriptor
+ * @param[in]   opt     option type
+ * @param[out]  value   pointer to store the option's value in
+ * @param[in]   max_len maximal amount of byte that fit into @p value
+ *
+ * @return              number of bytes written to @p value
+ * @return              `< 0` on error, 0 on success
+ */
+int netdev_get_pass(netdev_t *dev, netopt_t opt, void *value, size_t max_len);
+
```

Comment:
extra newline

Reply:
Also for the other comments, I'll give this whole thing a large cleanup as soon as I have time, probably not tomorrow, but somewhere this week.

Reply:
And fixed now :)